File: leetcode/0153-find-minimum-in-rotated-sorted-array/solution.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        l = 0
        r = n - 1
        if nums[l] < nums[r]:
            return nums[l]
        if n <= 2:
            return min(nums)

        while l < r:
            mid = l + (r - l) // 2
            if nums[mid] > nums[mid + 1]:
                return nums[mid + 1]
            if nums[l] < nums[mid]:
                l = mid + 1
            else:
                r = mid

        return nums[mid]

File: leetcode/0153-find-minimum-in-rotated-sorted-array/solution.java
class Solution {
    public int findMin(int[] nums) {
        int s = nums.length;

        if (s == 1) {
            return nums[0];
        }
        if (nums[0] < nums[s-1]) {
            return nums[0];
        }

        int l = 0;
        int r = s - 1;
        int mid = l + (r - l) / 2;

        while(l <= r) {
            mid = l + (r - l) / 2;
            if (nums[mid] > nums[mid + 1]) {
                return nums[mid + 1];
            }
            if (nums[mid] > nums[l]){
                l = mid;
            }
            else {
                r = mid;
            }
        }
        return nums[mid];
    }
}
